### PR TITLE
docs: highlight access to example configuration files

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
- Adds a new _Example configuration files_ section to the _Deploying Strimzi_ guide to describe what examples we provide and how to locate them.
- Updates references to example configuration files w/ links in the _Deploying Strimzi_, _Using Strimzi_ and _Strimzi Overview_ guides
- Removes files: con-product-downloads.adoc (legacy) and metrics/con_metrics-alertmanager-options.adoc (content incorporated into _Deploying Alertmanager_ procedure).

NOTE: New section does not mention example S2I files that  are still available

NOTE: Dashboard metrics lists weren't updated. Out of scope. 

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Looks good to me. Thanks.

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// deploying/assembly-deploy-tasks.adoc
+
+[id='deploy-examples-{context}']
+= Example configuration files
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:6]: _Suggested change:_

"Example configuration and deployment files"

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+[role="_abstract"]
+Use the example configuration files provided with Strimzi to deploy Kafka components and monitor your deployment.
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:9]:
```suggestion
Use the example configuration and deployment files provided with Strimzi to deploy Kafka components with different configurations and monitor your deployment.
```

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+Example configuration files for custom resources contain key properties and values, which you can extend with additional supported configuration properties for your own deployment.
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:10]: _Suggested change:_

key properties > important properties

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+<4> Cluster Operator template configuration for deploying Kafka components. Each supported Kafka component has a template configuration file. Includes templates for an ephemeral or persistent Kafka deployment.
+<5> Authentication and authorization configuration for Kafka components. Includes example configuration for TLS and SCRAM-SHA-512 authentication. The Keycloak example includes `Kafka` custom resource configuration and a Keycloak realm specification. You can use the example to try Keycloak authorization services.
+<6> `Kafka` custom resource configuration for a deployment of Mirror Maker. Includes example configuration for replication policy and synchronization frequency.
+<7> xref:ref-metrics-config-files-{context}[Metrics configuration], including Prometheus installation and Grafana dashboard files.
+<8> `Kafka` custom resource configuration for a deployment of Kafka. Includes example configuration for an ephemeral or persistent single or multi-node deployment.
+<9> `KafkaRebalance` custom resource configuration for a deployment of Cruise Control. Includes example configuration to use the default or main optimization goals.
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:62]:
```suggestion
<9> `Kafka` custom resource with a deployment configuration for Cruise Control. `KafkaRebalance` custom resources to generate optimizations proposals from Cruise Control, including example configurations to use the default or user optimization goals.
```

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+├── metrics <7>
+├── kafka <8>
+├── cruise-control <9>
+├── connect <10>
+└── bridge <11>
+--
+<1> `KafkaUser` custom resource configuration, which is managed by the User Operator.
+<2> `KafkaTopic` custom resource configuration, which is managed by Topic Operator.
+<3> Topic Operator template configuration for creating topics.
+<4> Cluster Operator template configuration for deploying Kafka components. Each supported Kafka component has a template configuration file. Includes templates for an ephemeral or persistent Kafka deployment.
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:57]: Sorry to ask a horrid question.

What is the difference between `examples/templates/connect-template.yaml` and `examples/connect/kafka-connect.yaml`?

File: documentation/modules/deploying/con-deploy-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// deploying/assembly-deploy-tasks.adoc
```
Comment (laidan6000) [inline documentation/modules/deploying/con-deploy-examples.adoc:3]:
```suggestion
// deploying/assembly-deploy-tasks-prep.adoc
```

**Review by laidan6000 — CHANGES_REQUESTED** [review]
Looks great. I made a few comments.

**Review by sknot-rh — APPROVED** [review]

File: documentation/modules/deploying/con-deploy-examples.adoc (hunk 5 above)
Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-examples.adoc:57]: Two different approaches. As far as I understand, OpenShift templates are like installing with Helm charts.
Is that right, @scholzj?

Comment (scholzj) [reply documentation/modules/deploying/con-deploy-examples.adoc:57]: Well, Helm is a way more powerful than the templates are. The OpenShift templates really just replace some keys with specified values. Maybe you should just not mention the templates at all and we should delete them, as the templates are crap and are anyway not really supported in OCP 4 that well.

Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-examples.adoc:57]: Thanks, @scholzj. One for the next sync up? Maybe we keep these callouts in for completeness until a decision is made?

Comment (PaulRMellor) [reply documentation/modules/deploying/con-deploy-examples.adoc]: As discussed, the _templates_ have been removed.

**Review by laidan6000 — APPROVED** [review]
Changes look good.

**Pull request merged**